protocol-specs/generators/multisignature_registration/index.js

```diff
@@ -0,0 +1,440 @@
+			'desk deposit crumble farm tip cluster goose exotic dignity flee bring traffic',
+		privateKey:
+			'ddc8e19d6697d6e5c1dacf6576a7169752810999918212afe14d3978b354f8aa4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
+		publicKey:
+			'4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
+		address: '3372320078773139180L',
+	},
+	optionalOne: {
+		passphrase:
+			'sugar object slender confirm clock peanut auto spice carbon knife increase estate',
+		privateKey:
+			'69aa94ea7ade3b7b08e277b18c1a590b2306ce5973ae8462b0b85122b180e89c57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
+		publicKey:
+			'57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
+		address: '7745870967079479156L',
+	},
+	optionalTwo: {
+		passphrase:
+			'faculty inspire crouch quit sorry vague hard ski scrap jaguar garment limb',
+		privateKey:
+			'ffed38380998a90a2af9501f10182bc2a07922448ab383575b1e34aeddfa5482fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
+		publicKey:
+			'fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
+		address: '7086965981385941478L',
+	},
+};
+
+const sortKeysDescending = publicKeys =>
+	publicKeys.sort((publicKeyA, publicKeyB) => {
+		// eslint-disable-next-line no-undef, new-cap
+		if (BigInt(`0x${publicKeyA}`) > BigInt(`0x${publicKeyB}`)) {
+			return 1;
+		}
+		// eslint-disable-next-line no-undef, new-cap
+		if (BigInt(`0x${publicKeyA}`) < BigInt(`0x${publicKeyB}`)) {
+			return -1;
+		}
+		return 0;
+	});
+
+const getId = transactionBytes => {
+	const transactionHash = hash(transactionBytes);
+	const bufferFromFirstEntriesReversed = getFirstEightBytesReversed(
+		transactionHash,
+	);
+	const transactionId = bufferToIntAsString(bufferFromFirstEntriesReversed);
+
+	return transactionId;
+};
+
+const assetToBytes = tx => {
+	const { mandatoryKeys, optionalKeys, numberOfSignatures } = tx.asset;
+	const mandatoryKeysBuffer = Buffer.from(mandatoryKeys.join(''), 'hex');
+	const optionalKeysBuffer = Buffer.from(optionalKeys.join(''), 'hex');
+
+	const assetBuffer = Buffer.concat([
+		intToBuffer(mandatoryKeys.length, 1),
+		mandatoryKeysBuffer,
+		intToBuffer(optionalKeys.length, 1),
+		optionalKeysBuffer,
+		intToBuffer(numberOfSignatures, 1),
+	]);
+	return assetBuffer;
+};
+
+const createSignatureObject = (txBuffer, account) => ({
+	signature: signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
+		account.passphrase,
+	),
+});
+
+const serializeBasicProperties = tx => {
+	const transactionTimestamp = Buffer.alloc(4);
+	transactionTimestamp.writeIntBE(tx.timestamp, 0, 4);
+	const buf = Buffer.concat([
+		Buffer.alloc(1, tx.type),
+		transactionTimestamp,
+		hexToBuffer(tx.senderPublicKey),
+		assetToBytes(tx),
+	]);
+
+	return buf;
+};
+
+const serializeMemberSignatures = (tx, txBuffer) => {
+	let txBufferCopy = Buffer.alloc(txBuffer.length);
+	txBuffer.copy(txBufferCopy);
+
+	tx.signatures.forEach(aSignature => {
```

Comment:
This serialization of the signatures is not as specified in LIP: 0x01 has to be added for each signature and 0x00 for every empty buffer in tx.signatures.

Reply:
Isn't that type of serialization for other types of transactions originating from a multisignature? This is multisig registration only

Reply:
this serialization has to be applied for every transaction, including transactions from "regular" accounts and multisig registration transactions.
That also means that for transactions from non-multisig accounts the 0x01 has to be prepended to the signature.

Reply:
Fixed. Good catch!